4 reports.sql

```diff
@@ -0,0 +1,42 @@
+--The reports I need are as following:
+-- 1) How many people are booked per flight as: count of people, flight number, departure airport, when the flight is departing and destination.
+select p.FlightNumber, P.DepartureAirport, p.TimeDeparting, p.ArrivalAirport, AmountOfPassengers = count(*)
+from Passenger p 
+group by p.FlightNumber, P.DepartureAirport, p.TimeDeparting, p.
+
+-- 2) Who isn't checked in for flights departing in the next week, in order to send them reminders to check in.
+select * 
+from Passenger p 
+where PassportIssueDate is null 
+and TimeDeparting < dateadd(week, 1, getdate())
+
+-- 3) How many flights are departing per day, and num of passengers we have on those flights.
+--was it possible to do this in one report?
+select DepartureDate = convert(varchar, p.TimeDeparting, 103), p.FlightNumber, NumPassengers = Count(*)
+from Passenger p 
+group by convert(varchar, p.TimeDeparting, 103), p.FlightNumber
+
+select DepartureDate = convert(varchar, p.TimeDeparting, 103), FlightsDeparting = count(distinct p.FlightNumber)
+from Passenger p
+group by convert(varchar, p.TimeDeparting, 103)
+
+-- 4) How many flights are departing per destination, and num of passengers we have on those flights, to know what route is attracts most people.
+--not sure how to do this in one report
```

Comment:
How about this?  Same idea as previous comment.
`select p.ArrivalAirport, NumberOfFlights = count(distinct p.FlightNumber), NumOfPassengers = count(*)
from Passenger p
group by p.ArrivalAirport`

Reply:
adjusted